scheduling-manager/src/main/java/com/amazonaws/blox/schedulingmanager/stream/processor/StreamProcessorHandler.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.schedulingmanager.stream.processor;
+
+import com.amazonaws.blox.schedulingmanager.stream.StreamProcessorApplication;
+import com.amazonaws.blox.schedulingmanager.stream.exception.InvalidRecordStructureException;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.DynamodbEvent;
+import com.amazonaws.services.lambda.runtime.events.DynamodbEvent.DynamodbStreamRecord;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+@Slf4j
+public class StreamProcessorHandler implements RequestHandler<DynamodbEvent, String> {
+
+  public static final ApplicationContext applicationContext =
+      new AnnotationConfigApplicationContext(StreamProcessorApplication.class);
+
+  @Override
+  public String handleRequest(final DynamodbEvent dynamodbEvent, final Context context) {
+
+    for (DynamodbStreamRecord record : dynamodbEvent.getRecords()) {
+      log.debug(record.toString());
+
+      final RecordProcessor recordProcessor = applicationContext.getBean(RecordProcessor.class);
+
+      try {
+        recordProcessor.process(record);
```

Comment:
Are there other ways in which record processing can fail, maybe non-fatally? If so, what can we do to re-process the record; is there some kind of retry/DLQ support in the DDB Streams world?

Either way, if this is trying out our approach to handling these streams, we should probably build some partial failure handling into the RecordProcessor interface.

Reply:
If the processing of the record fails, the entire batch that's passed into a Lambda fails and Lambda retries the batch until successful or expired.

Reply:
Does that mean we need to ensure that the underlying RecordProcessor is idempotent? Seems like it could be asked to process the same record many times...

Reply:
Yes, it's described in the [design doc.](https://github.com/blox/blox/blob/dev/docs/daemon_infrastructure.md)